src/Navigation/Dropdown/Dropdown.tsx

```diff
@@ -17,231 +17,180 @@ import {
-      hoverToOpen, // eslint-disable-line @typescript-eslint/no-unused-vars
-      onChange, // eslint-disable-line @typescript-eslint/no-unused-vars
-      ...defaultProps
-    } = this.props;
-
-    const { isOpen, dropdownLabel, cursor } = this.state;
-
-    const LeftIcon = leftIcon;
-
-    return (
-      <DropdownContainer
-        className={classNames('aries-dropdown', className)}
-        tabIndex={0}
-        onClick={this.handleOpen}
-        onMouseEnter={this.hoverOpen}
-        onMouseLeave={this.hoverClose}
-        onBlur={this.handleClose}
-        onKeyDown={this.handleKeyDown}
-        role="menuitem"
-        aria-label={dropdownLabel}
-        aria-haspopup="true"
-        {...defaultProps}
-      >
-        <DropdownWrapper className="dropdown-contentwrapper" tabIndex={-1}>
-          <DropdownHeader
-            className="dropdown-content"
-            isOpen={isOpen}
-            disabled={disabled}
-            showHoverLine={showHoverLine}
-            showFullWidth={showFullWidth}
-          >
-            {LeftIcon && (
-              <LeftIcon color={!disabled ? iconDefaultColor : '#777777'} />
-            )}
-            <span>{itemElement ? itemElement : dropdownLabel}</span>
-            <IconWrapper isOpen={isOpen}>
-              <ArrowDownIcon color={!disabled ? iconDefaultColor : '#777777'} />
-            </IconWrapper>
-          </DropdownHeader>
-          <DropdownBody
-            className="dropdown-listbox"
-            role="listbox"
-            aria-hidden={!isOpen && true}
-            onClick={(e: React.MouseEvent) => e.stopPropagation()}
-            open={isOpen}
-            dropDownPlacement={dropDownPlacement}
-            noLineBreak={noLineBreak}
-            showFullWidth={showFullWidth}
-            showHoverLine={showHoverLine}
-            ref={this.dropdownBodyRef}
-          >
-            {React.Children.map(
-              children,
-              (item: React.ReactElement<DropdownItemProps>, index) => {
-                const dropDownItemClassName = item.props.className || '';
-                return (
-                  <DropdownItemWrapper
-                    className={classNames(
-                      { active: cursor === index },
-                      `${dropDownItemClassName}`
-                    )}
-                    role="option"
-                    data-value={item.props.value}
-                    key={item.key}
-                    onMouseDown={this.handleClickItem(item.props.onClick)}
-                    onMouseEnter={() => this.handleMouseEnter(index)}
-                    tabIndex={0}
-                    showFullWidth={showFullWidth}
-                  >
-                    {item.props.children}
-                  </DropdownItemWrapper>
-                );
-              }
-            )}
-          </DropdownBody>
-        </DropdownWrapper>
-      </DropdownContainer>
-    );
-  }
-}
+  React.useLayoutEffect(() => {
```

Comment:
why need to use `useLayoutEffect`?

Reply:
When I'm replacing `onComponentDidMount` I'm defaulting to `useLayoutEffect` because I had some trouble with tests in the past: https://glints.slack.com/archives/C956ZDAP7/p1584325219041900

Reply:
hmm, because this effect is not related to update dom elements, so useEffect is ok.